Add composable schema docs (#306)
authzed/docs: pages/spicedb/modeling/composable-schemas.mdx

import { Callout, Tabs } from 'nextra/components'

# Composable Schemas (Preview)

<Callout type="info">

> This is a preview feature whose functionality is subject to change before

Reviewer: ```suggestion
This preview feature's functionality may change before general release.
```

general release.
</Callout>

> In zed version v0.27.0, we introduced a compilation command:

Reviewer: ```suggestion
In zed version v0.27.0, we introduced a schema compilation command:
```

```
zed preview schema compile some-schema.zed
```

There are three new pieces of syntax: [import statements](#import-statements), [partial declarations](#partial-declarations), and [partial references](#partial-references).

This is a simple schema that demonstrates all three features:

<Tabs items={["root.zed", "subjects.zed"]}>
<Tabs.Tab>

```zed
import "./subjects.zed"

partial view_partial {
    relation user: user
    permission view = user
}

definition resource {
    ...view_partial

    relation organization: organization
    permission manage = organization
}
```

</Tabs.Tab>
<Tabs.Tab>

```zed
definition user {}
definition organization {}
```

</Tabs.Tab>
</Tabs>

Compiling the above with `zed preview schema compile root.zed` will produce an output schema that looks like:

```zed
definition user {}

definition organization {}

definition resource {
    relation user: user
    permission view = user

    relation organization: organization
    permission manage = organization
}
```

> This compilation step introduces a new set of features that allow you to break down your schema,

Reviewer: ```suggestion
This compilation step provides new features that help you modularize your schema, making organization and collaboration easier.
```
